service-base/src/test/java/org/eclipse/hono/service/credentials/AbstractCompleteCredentialsServiceTest.java

```diff
@@ -51,12 +50,10 @@ public void testAddCredentialsRefusesDuplicateRegistration(final TestContext ctx
                 .put(CredentialsConstants.FIELD_AUTH_ID, "myId")
                 .put(CredentialsConstants.FIELD_TYPE, "myType")
                 .put(CredentialsConstants.FIELD_SECRETS, new JsonArray());
-        final Async add = ctx.async();
-        getCompleteCredentialsService().add("tenant", payload2, ctx.asyncAssertSuccess(s -> {
-            assertThat(s.getStatus(), is(HttpURLConnection.HTTP_CONFLICT));
-            add.complete();
-        }));
-        add.await();
+        getCompleteCredentialsService().add("tenant", payload2, ctx.succeeding(s -> ctx.verify(() -> {
+            assertEquals(HttpURLConnection.HTTP_CONFLICT, s.getStatus());
+            ctx.completeNow();
```

Comment:
it looks strange to me that you would need to explicitly invoke this method. Have you checked if it works without?

Reply:
the `ctx.completeNow();` method ? Yes, I agree with you that it's a bit odd. 
I went to re-read the vertx doc and send you a link but it look like when the test expect a success I could use `ctx.completing()` which complete the test context on a success. I am going to try it !

Reply:
So as @jbtrystram explained, with vertx it is necessary to let the async test cases know that the test is actually completed. As the idea is that the unit tests may still be running when it exited the method annotated with `@Test`.

One thing to note here, the call to `completeNow()` actually completes the test context. If the test has multiple steps, you need to use `Checkpoint` as done in the other tests.